app/processor/string.go

```diff
@@ -0,0 +1,198 @@
+			case urlEncodingFlag:
+				return m[identifierMatchIndex], URL
+			case plainEncodingFlag:
+				return m[identifierMatchIndex], Plain
+			}
+		}
+	}
+	return data, defaultEncoding
+}
+
+func (j *JobStringProcessor) ProcessJobString(data string, options ...*Options) string {
+	matches := reExpression.FindAllStringSubmatch(data, -1)
+	option := getOption(options...)
+
+	// search for ${...} expressions in the data string
+	for _, submatch := range matches {
+		// content of the ${...} expression, without the brackets
+		match := submatch[1]
+		// now it is possible to override the encoding with flags: ${encoding=plain;identifier}
+		// if encoding is supplied via flags, it takes precedence, we return the match
+		// without the flags, otherwise fail back to original match and encoding
+		match, option.Encoding = getEncodingFromData(match, option.Encoding)
+		defaultValue := submatch[2]
+		if strings.HasPrefix(match, workflowInputVariable) &&
+			len(match) > len(workflowInputVariable) {
+			// Replace ${workflow.input.KEY} with the KEY input variable
+			paramName := match[len(workflowInputVariable):]
+			found := false
+			for _, param := range j.job.InputParameters {
+				if param.Name == paramName {
+					value := param.Value
+					if option.Encoding == URL {
+						value = url.QueryEscape(value)
+					}
+					data = strings.ReplaceAll(data, submatch[0], value)
+					found = true
+					break
+				}
+			}
+			if !found {
+				data = strings.ReplaceAll(data, submatch[0], defaultValue)
+			}
+		} else if strings.HasPrefix(match, workflowEnvVariable) &&
+			len(match) > len(workflowEnvVariable) {
+			// Replace ${env.KEY} with the KEY environment variable
+			envName := match[len(workflowEnvVariable):]
+			envValue := os.Getenv(envName)
+			if envValue == "" {
+				envValue = defaultValue
+			}
+			data = strings.ReplaceAll(data, submatch[0], envValue)
+		} else if output := reExpressionOutput.FindStringSubmatch(match); len(output) > 0 {
+			// Replace ${TASK_NAME.json.JSONPATH} with the value of the JSONPATH expression from the
+			// JSON output of the previous task with name TASK_NAME. If the output is not a valid
+			// JSON or the JSONPATH does not resolve to a value, replace with empty string
+			for _, result := range j.job.Results {
+				if result.Name == output[1] {
+					varKey := output[2]
+					var output string
+					if result.Type == model.TaskTypeHTTP {
+						output = result.HTTPResponse.Body
+					} else if result.Type == model.TaskTypeCLI {
+						output = result.CLI.Output
+					} else {
+						continue
+					}
+					varValue := gojsonq.New().FromString(output).Find(varKey)
+					if varValue == nil {
+						varValue = ""
+					}
+					varValueString, err := utils.ConvertAnythingToString(varValue)
+					if err == nil {
+						if varValueString == "" {
+							varValueString = defaultValue
+						}
+						data = strings.ReplaceAll(data, submatch[0], varValueString)
+					}
+					break
+				}
+			}
+		}
```

Comment:
Finally, this applies the option to all the replacements above.
```suggestion
		}
		if option.Encoding == URL {
			value = url.QueryEscape(value)
		}
		data = strings.ReplaceAll(data, submatch[0], value)
```